### ISSUE TITLE
discord >= 0.0.10 is overlinked with libcups.so.2

### ISSUE THREAD

**Issue by glebfm** [crmarsh/discord-linux-bugs#131]
````
$ ldd --unused discord-canary/DiscordCanary
Unused direct dependencies:
        /usr/lib64/libcups.so.2
$ ldd DiscordCanary | sed -e 's/(0x................)//' > o
$ patchelf --remove-needed libcups.so.2 DiscordCanary
$ ldd DiscordCanary | sed -e 's/(0x................)//' > oo
$ diff -u o oo | grep '^[-+]'
[...]
-       libcups.so.2 => /usr/lib64/libcups.so.2
-       libgssapi_krb5.so.2 => /lib64/libgssapi_krb5.so.2
-       libavahi-common.so.3 => /usr/lib64/libavahi-common.so.3
-       libavahi-client.so.3 => /usr/lib64/libavahi-client.so.3
-       libkrb5.so.3 => /lib64/libkrb5.so.3
-       libk5crypto.so.3 => /lib64/libk5crypto.so.3
-       libcom_err.so.2 => /lib64/libcom_err.so.2
-       libkrb5support.so.0 => /lib64/libkrb5support.so.0
-       libkeyutils.so.1 => /lib64/libkeyutils.so.1

**Comment by crmarsh** [comment]
Does this cause a problem, or is it just unclean?

**Comment by OvermindDL1** [comment]
It should not cause issues if the person has cups installed, but if they do not then it would likely not load (admittedly that would be a rare case for someone not to have it...).

**Comment by DragoonAethis** [comment]
Actually not so rare - you can easily not have CUPS installed at all if you don't have any printers and your distro allows you to skip that (eg Gentoo, Arch with dummy packages) while installing other packages.

**Comment by OvermindDL1** [comment]
True yeah, so it would be good to clean it out...

**Comment by crmarsh** [comment]
Wait, maybe I misunderstood, is the bug that we depend on libcups, it's that there are redundant dependencies?

If the bug is that we depend on cups and you might not have it, I'm not going to fix that, just install cups.

**Comment by DragoonAethis** [comment]
Yeah, Discord shouldn't depend on CUPS - it's for printing. AFAIK disabling printing in your Electron build should do the trick.

**Comment by crmarsh** [comment]
I'm still unclear, why should I be spending time on this?

**Comment by DragoonAethis** [comment]
- If you have CUPS installed, it just loads the libcups library and keeps it in memory. Just unclean, and if you ever call libcups (for any reason - saving a page as a PDF, for example) you also get to wait until it connects to the CUPS server, queries drivers, wakes up network devices if there are any set up...
- If you don't have CUPS installed, Discord won't start, even though it ~shouldn't~doesn't use CUPS for anything.

**Comment by crmarsh** [comment]
FWIW, this is an [upstream issue](https://github.com/electron/libchromiumcontent/issues/134).

**Comment by Turmfalke2** [comment]
If you don't feel like installing printer software to use an VoIP client you can remove the dependency with "patchelf --remove-needed libcups.so.2 Discord"
